src/Uno.UI/UI/Xaml/DependencyObjectStore.cs

```diff
@@ -1634,13 +1634,8 @@ internal IEnumerable<ResourceDictionary> GetResourceDictionaries(
 					{
 						yield return fe.Resources;
 					}
-
-					candidate = fe.Parent as FrameworkElement;
-				}
-				else
-				{
-					candidate = VisualTreeHelper.GetParent(candidate) as DependencyObject;
 				}
+				candidate = VisualTreeHelper.GetParent(candidate) as DependencyObject;
```

Comment:
Let's make sure to add tests for this, this is a pretty low level change :)

Reply:
added a test